backend/modules/invoices.js

```diff
@@ -2,23 +2,43 @@ import { v4 as uuidv4 } from 'uuid';
 import { Router } from 'express';
 import pool from '../utils/database.js';
 import fs from 'fs';
+import express from 'express';
+import fileUpload from 'express-fileupload';  
+import bodyParser from 'body-parser';
 
 const router = Router();
+const app = express();
 
+// Middleware
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(fileUpload());  
+
+// Check if the role is valid (EC, CC, or JC)
+function checkRole(req, res, next) {
```

Comment:
You can remove this as well, ig this function already exists somewhere in the code.